http.ts

```diff
@@ -0,0 +1,129 @@
+import {log} from "./util";
+
+import { pubInternal, sub } from "./dispatch";
+import { main as pb } from "./msg.pb";
+
+const servers = new Map<number, HttpServer>();
+const dec = new TextDecoder("utf8");
+const enc = new TextEncoder();
+
+export function initHttp() {
+  sub("http", (payload: Uint8Array) => {
+      const msg = pb.Msg.decode(payload);
+      const id = msg.httpServerId;
+      const s = servers.get(id);
+      s.onMsg(msg);
+  });
+}
+
+export interface RequestOptions {
+    method?: string;
+    referrer?: string;
+    mode?: string;
+    credentials?: string;
+    redirect?: string;
+    integrity?: string;
+    cache?: string;
+}
+
+export class Request {
+    readonly defaultMethod = "GET";
+    path: string;
+    method: string;
+    // tslint:disable-next-line: no-any
+    body: any | string;
+    constructor(url: string, opts?: RequestOptions) {
+      if (opts == null) {
```

Comment:
@matiasinsaurralde I think you meant `undefined`.

`?` in Typescript is not `null` but `undefined`.

Reply:
`x == null` is often used to testing against both null and undefined. 
tslint makes exceptions for triple equals in this case.

I think it's ok.

Reply:
@ry fair enough that makes sense I just prefer to the code to mirror what the type system says

Related to https://github.com/ry/deno/issues/234#issuecomment-396042429